### PR TITLE
Add complex formula API

### PR DESCRIPTION
**Pull request by jheander**

### CODE REVIEW

**Review by lars-timezynk — COMMENTED** [review]
Can you require operands and operator to always be set? Shouldn't they be optional when using recursion?

Maybe have a node object and a separate leaf object?

**Comment by jheander** [comment]
That was kind of my intention in wrapping `operands` and `operator` inside the `formula` key. So a leaf object will have `formula` nil, and set `value` or `field-id` instead

File: spec/components/schemas/ArticleFormula.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      operands:
+        type: array
+        items:
+          $ref: "#/components/schemas/ArticleFormula"
+      operator:
+        type: string
+        enum:
+          - +
+          - "-"
+          - "*"
```
Comment (sebastianjonasson) [inline spec/components/schemas/ArticleFormula.yaml:23]: Unquote options
```suggestion
          - -
          - *
```

**Review by sebastianjonasson — COMMENTED** [review]
Considering the case below the operator doesn't really fill a purpose when `operands.length` is 1. In practice I guess the `operator` will be omitted from this case, but for completion maybe we should leave it optional and default to `AVG`, or have a `noop` operator?


Example `30 + 20 - 10`

```javascript
{
  "formula": {
    "operands": [
      {
        "formula": {
          "operands": [
            {
              "value": 30
            },
            {
              "value": 20
            }
          ],
          "operator": "+"
        }
      },
      {
        "formula": {
          "operands": [
            {
              "value": 10
            }
          ],
          "operator": "?" // does operator make sense  here?
        }
      }
    ],
    "operator": "-"
  }
}
```

File: spec/components/schemas/ArticleFormula.yaml (hunk 1 above)
Comment (jheander) [reply spec/components/schemas/ArticleFormula.yaml:23]: I had some issues with "- -" in yaml, but maybe that was caused by something else :thinking:

**Comment by jheander** [comment]
Regarding operands with single entry, maybe it is still good to force the client to take a stand a specify an operator. The most common operator is probably "+", so we could also set that as default since it has a clear interpretation for any number of operands.

File: spec/components/schemas/ArticleFormula.yaml (hunk 1 above)
Comment (sebastianjonasson) [reply spec/components/schemas/ArticleFormula.yaml:23]: Ah, I get it. And also it actually renders correctly in the docs. So I guess it doesn't matter.

**Pull request merged**